### PR TITLE
Buttery vacation: clean up glitch pro data management

### PR DESCRIPTION
**Pull request by modernserf**
## Links
* Remix link: https://buttery-vacation.glitch.me/
* Issue-tracker link: https://app.clubhouse.io/glitch/story/7125/fix-subscription-cancellation-page

## Changes:
* store state in context, so only one request is made for header, footer & pages
* share subscribe functions between pricing and settings page
* update subscription state when cancelling
* rename to `useGlitchPro`

### CODE REVIEW

File: src/components/account-settings-overlay/subscription-settings.js [hunk 1]
```diff
@@ -3,69 +3,46 @@ import { Button, Loader } from '@fogcreek/shared-components';
...
-      const { id: sessionId } = data;
-      stripe.redirectToCheckout({ sessionId });
-    } catch (err) {
-      // TODO decide what kind of error handling we need here
-      console.log(err);
-    }
-  }
-
-  async function cancel() {
+  const cancelWithWait = async () => {
```
Comment (sarahzinger) [inline src/components/account-settings-overlay/subscription-settings.js]: just curious whats the `withWait` part here indicate to folks?

File: src/state/glitch-pro.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  const [loaded] = useScript(stripeJS);
+  useEffect(() => {
+    if (loaded) {
+      setStripe(window.Stripe('pk_test_WIJxKl0T1xT8zSb2StHu8zlo'));
+    }
+  }, [loaded]);
+  return stripe;
+}
+
+function useGlitchProState() {
```
Comment (sarahzinger) [inline src/state/glitch-pro.js]: would it make sense at this point to add some tests? We don't have a lot but I don't think anything technically is stopping us from making that happen.

File: src/components/account-settings-overlay/subscription-settings.js (hunk 1 above)
Comment (modernserf) [reply src/components/account-settings-overlay/subscription-settings.js]: maybe `disabledButtonAndCancel`?

Comment (sarahzinger) [reply src/components/account-settings-overlay/subscription-settings.js]: ah ok I see what you mean, yeah how you have is fine! just wanted to know if this is like a common pattern I should learn about or something

**Pull request merged**